src/proto/Nano.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+package TW.Nano.Proto;
+option java_package = "wallet.core.jni.proto";
+
+// Input data necessary to create a signed transaction.
+message SigningInput {
+    // Private key
+    bytes private_key = 1;
+
+    // Optional parent block hash
+    bytes parent_block = 2;
+
+    oneof link_oneof {
+        // Hash of a block to receive from
+        bytes link_block = 3;
+        // Recipient address to send coins to
+        string link_recipient = 4;
+    }
+
+    // Representative address
+    string representative = 5;
+
+    // Account balance (128-bit number, big-endian encoded)
+    bytes balance = 6;
```

Comment:
Manipulating Nano balances requires uint128 support, in most languages it comes via 3rd party libraries (Swift, JS and Java all have them). Currently the Nano::Proto::SigningInput takes the encoded bytes value (from said libraries), but it might be worth considering taking a string representation of the number instead. That way the users wouldn't have to figure out if they're encoding the balance value correctly in big-endian or incorrectly in little-endian. (Also it would make the unit-tests look somewhat nicer)

Let me know what are your feelings regarding this: passing in a 16-bytes of data, or passing in a string (that wallet-core parses).

Reply:
Yes, string might be better, @alejandro-isaza how do you think?

Reply:
Yes, string works.